src/systems/userinput/userinput.js

```diff
@@ -348,17 +355,34 @@ AFRAME.registerSystem("userinput", {
       this.masks = computeMasks(this.sortedBindings);
     }
 
-    this.prevActiveSets.clear();
-    for (const item of this.activeSets) {
-      this.prevActiveSets.add(item);
+    this.prevActiveSets.length = 0;
+    for (let i = 0; i < this.activeSets.length; i++) {
+      const item = this.activeSets[i];
+
+      if (!this.prevActiveSets.includes(item)) {
+        this.prevActiveSets.push(item);
+      }
     }
     resolveActionSets();
-    for (const { set, value } of this.pendingSetChanges) {
-      this.activeSets[value ? "add" : "delete"](set);
+    for (let i = 0, l = this.pendingSetChanges.length; i < l; i += 2) {
+      const set = this.pendingSetChanges[i];
+      const value = this.pendingSetChanges[i + 1];
+
+      if (value) {
+        if (!this.activeSets.includes(set)) {
+          this.activeSets.push(set);
+        }
+      } else {
+        const idx = this.activeSets.indexOf(set);
+
+        if (idx > -1) {
+          this.activeSets.splice(idx, 1);
+        }
+      }
     }
     const activeSetsChanged =
-      this.prevActiveSets.size !== this.activeSets.size ||
-      intersection(this.prevActiveSets, this.activeSets).size !== this.activeSets.size;
+      this.prevActiveSets.length !== this.activeSets.length ||
+      intersectionSize(this.prevActiveSets, this.activeSets) !== this.activeSets.length;
```

Comment:
We only care if there exists something in `prevActiveSets` that isn't in `activeSets` or vice versa, so we could replace `intersectionSize` with a function `isDifferent` that would could terminate as soon as it finds a difference. These collections are so small though (<10 items) that it doesn't actually matter.